application.py

```diff
@@ -113,9 +132,10 @@ def show_qr_code(self):
             if self.current_video_index > 0:
                 current_video_data = self.video_data[self.current_video_index - 1]
                 video_id = current_video_data['id']
-                equip_id = service.get_param_from_config('config.ini', 'PN')
+                equip_id = self.service.get_param_from_config('config.ini', 'PN')
                 equip_ip = "192.168.1.1"
-                url = f"https://link.olhar.media/?golink=1&equipid={equip_id}&videoid={video_id}&equipip={equip_ip}&gpslat=40&gpslon=40"
+                lat_and_lon = self.service.get_lat_lon()
```

Comment:
This may take time. Not for this PR, but you will need to either get coordinates just once and use it for the whole session (if the customer prefers that) or prepare the URL in parallel while the video is playing. And you need to have a fallback giving the old coordinates in case GPS is not available.